core/src/main/java/org/infinispan/util/concurrent/locks/LockUtil.java

```diff
@@ -27,18 +28,14 @@ public static Ownership getLockOwnership(Object key, ClusteringDependentLogic cl
       Object keyToCheck = key instanceof DeltaCompositeKey ?
             ((DeltaCompositeKey) key).getDeltaAwareValueKey() :
             key;
-      if (clusteringDependentLogic.localNodeIsPrimaryOwner(keyToCheck)) {
-         return Ownership.PRIMARY;
-      } else if (clusteringDependentLogic.localNodeIsOwner(keyToCheck)) {
-         return Ownership.BACKUP;
-      } else {
-         return Ownership.NON_OWNER;
-      }
+      DistributionInfo distributionInfo = clusteringDependentLogic.getCacheTopology().getDistribution(keyToCheck);
+      return distributionInfo.isPrimary() ? Ownership.PRIMARY :
```

Comment:
I was hoping to avoid the discussion of read vs write primary by having a single `isPrimary()` in there :)